packages/commons/src/message-queue/record.ts

```diff
@@ -0,0 +1,202 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
+ */
+import { Job, Processor, Queue, Worker } from 'bullmq'
+import { PlainToken } from '../http'
+import { logger } from '../logger'
+import {
+  BirthRegistration,
+  DeathRegistration,
+  EVENT_TYPE,
+  MarriageRegistration,
+  SupportedPatientIdentifierCode,
+  WaitingForValidationRecord
+} from '../types'
+
+type ExternalValidationPayload = {
+  record: WaitingForValidationRecord
+  token: PlainToken
+}
+export type RecordValidatedPayload = {
+  token: PlainToken
+  trackingId: string
+  registrationNumber: string
+  recordId: string
+  identifiers: {
+    type: SupportedPatientIdentifierCode
+    value: string
+  }[]
+}
+
+type Payload =
+  | {
+      event: EVENT_TYPE.BIRTH
+      payload: BirthRegistration
+      token: PlainToken
+    }
+  | {
+      event: EVENT_TYPE.DEATH
+      payload: DeathRegistration
+      token: PlainToken
+    }
+  | {
+      event: EVENT_TYPE.MARRIAGE
+      payload: MarriageRegistration
+      token: PlainToken
+    }
+
+export function useExternalValidationQueue(redisHost: string) {
+  const queue = new Queue<ExternalValidationPayload | RecordValidatedPayload>(
+    'external-validations',
+    {
+      connection: { host: redisHost, port: 6379 }
+    }
+  )
+
+  async function sendForExternalValidation(payload: ExternalValidationPayload) {
+    await queue.waitUntilReady()
+
+    await queue.add('send-to-external-validation', payload, {
+      attempts: Number.MAX_SAFE_INTEGER,
+      removeOnComplete: true,
+      backoff: {
+        type: 'exponential',
+        delay: 1000
+      }
+    })
+  }
+  async function recordValidated(payload: RecordValidatedPayload) {
+    await queue.waitUntilReady()
+
+    await queue.add('record-validated-externally', payload, {
+      attempts: Number.MAX_SAFE_INTEGER,
+      removeOnComplete: true,
+      backoff: {
+        type: 'exponential',
+        delay: 1000
+      }
+    })
+  }
+
+  return {
+    recordValidated,
+    sendForExternalValidation
+  }
+}
+export function useRecordQueue(redisHost: string) {
+  const queue = new Queue<Payload>('records', {
+    connection: { host: redisHost, port: 6379 }
```

Comment:
allow configuring the port? it's usually allowed in the constants files

Reply:
Todos updated

Reply:
Updated in the confirm/rejectRegistration PR.